### PR TITLE
Make StringUtils.split faster

### PR DESCRIPTION
**Pull request by greenrobot-team**
Now seems to be faster than JVM implementation.
This pull request will have some conflicts with #13, so better to start from #13.

### CODE REVIEW

**Comment by greenrobot-team** [comment]
The general problem, that latest JVM implementation has now a check, if delimiter is only one character. In that case, regexp is not involved. The same is with latest Android implementation. However I found, that Android implementation is even faster than JVM's. So the latest commit is based on Android implementation from KitKat, but with some simplifications, so it is also a bit faster then on Android.

The current implementation compared to JVM:
- 50% faster for tiny string (like "John Doe")
- 20% faster for any string that is like "The quick brown fox jumps over the lazy dog" or any longer string
- consumes twice or more (depending on list resize count) less memory during it's run, because no additional list created

For Android it is faster by ~10% for tiny strings, and almost the same for longer strings.

**Comment by greenrobot-team** [comment]
Again took the liberty to rebase against latest `master` and squash the commits into one.

**Comment by greenrobot-team** [comment]
LGTM, merging.

Some results from Nexus 5/Android 6.0.1, values in seconds:
```
						Std     Lib
LongHashMap (Dynamic)	0.353	0.271
LongHashMap (Prealloc)	0.348	0.246
LongHashSet (Dynamic)	0.367	0.264
LongHashSet (Prealloc)	0.349	0.243
PipelineStream			12.229	0.015
StringSplit (Short)		0.862	1.166
StringSplit (Long)		0.888	1.141
```

File: java-essentials/src/main/java/org/greenrobot/essentials/StringUtils.java [hunk 1]
```diff
@@ -36,26 +36,28 @@ public class StringUtils {
             '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'
     };
 
-    /** Splits a String based on a single character, which is usually faster than regex-based String.split(). */
+    /**
+     * Splits a String based on a single character, which is usually faster than regex-based String.split().
+     * NOTE: split("AA;BB;;", ';') == ["AA", "BB", "", ""], this may be different from String.split()
+     *  */
     public static String[] split(String string, char delimiter) {
```
Comment (TG5069) [inline java-essentials/src/main/java/org/greenrobot/essentials/StringUtils.java:43]: ![ZJM.postman_globals.txt](https://github.com/user-attachments/files/18257554/ZJM.postman_globals

**Review by TG5069 — APPROVED** [review]

**Pull request merged**